tests/unit/models/QuestionTest.php

```diff
@@ -44,4 +44,94 @@ public function testGetQuestionThemeOnSavedQuestion()
         $this->assertNotEmpty($questionTheme);
         $this->assertEquals("bootstrap_buttons_multi", $questionTheme->name);
     }
+
+    /**
+     * Test validating a question theme name
+     * when question type is set to null.
+     */
+    public function testValidateQuestionThemeNameNoType()
+    {
+        $question = new Question();
+        $question->type = null;
+
+        $result = $question->questionThemeNameValidator();
+        $this->assertNull($result, 'An invalid validation result was expected since the question type was set to null.');
+    }
+
+    /**
+     * Test validating a question theme name
+     * when evaluating a child question.
+     */
+    public function testValidateChildQuestionThemeName()
+    {
+        $question = new Question();
+        $question->parent_qid = 1;
+
+        $result = $question->questionThemeNameValidator();
+        $this->assertNull($result, 'An invalid validation result was expected since a child question is being validated.');
+    }
+
+    /**
+     * Test getting a default theme name from type
+     */
+    public function testGetDefaultQuestionThemeNameFromType()
+    {
+        $question = new Question();
+
+        $themeName = $question->questionThemeNameValidator();
+        $this->assertSame('longfreetext', $themeName, 'Incorrect default question theme name.');
+    }
+
+    /**
+     * Test getting a default theme name from an invalid type.
+     */
+    public function testGetDefaultQuestionThemeNameFromInvalidType()
+    {
+        $question = new Question();
+        $question->type = 'Test';
+
+        $themeName = $question->questionThemeNameValidator();
+        $this->assertNull($themeName, 'An invalid validation result was expected since "Z" question type does not exist.');
```

Comment:
'An invalid validation result was expected since "Test" question type does not exist.'